src/Base/UnitsSchemaCentimeters.cpp

```diff
@@ -22,56 +22,40 @@
 
 
 #include "PreCompiled.h"
-#ifdef __GNUC__
-#include <unistd.h>
+#ifndef _PreComp_
+#include <algorithm>
+#include <array>
+#include <utility>
 #endif
 
-#include <QString>
-
 #include "UnitsSchemaCentimeters.h"
 
-
 using namespace Base;
 
-
-QString UnitsSchemaCentimeters::schemaTranslate(const Base::Quantity& quant,
-                                                double& factor,
-                                                QString& unitString)
+std::string UnitsSchemaCentimeters::schemaTranslate(const Base::Quantity& quant,
+                                                    double& factor,
+                                                    std::string& unitString)
 {
-    Unit unit = quant.getUnit();
-    if (unit == Unit::Length) {
-        // all length units in centimeters
-        unitString = QString::fromLatin1("cm");
-        factor = 10.0;
-    }
-    else if (unit == Unit::Area) {
-        // all area units in square meters
-        unitString = QString::fromLatin1("m^2");
-        factor = 1000000.0;
-    }
-    else if (unit == Unit::Volume) {
-        // all area units in cubic meters
-        unitString = QString::fromLatin1("m^3");
-        factor = 1000000000.0;
-    }
-    else if (unit == Unit::Power) {
-        unitString = QString::fromLatin1("W");
-        factor = 1000000;
-    }
-    else if (unit == Unit::ElectricPotential) {
-        unitString = QString::fromLatin1("V");
-        factor = 1000000;
-    }
-    else if (unit == Unit::HeatFlux) {
-        unitString = QString::fromLatin1("W/m^2");
-        factor = 1.0;
-    }
-    else if (unit == Unit::Velocity) {
-        unitString = QString::fromLatin1("mm/min");
-        factor = 1.0 / 60;
+    std::array<std::pair<Unit, std::pair<std::string, double>>, 7> unitSpecs {{

```

Comment:
I think this can be declared as static constexpr or do I miss sometihng.

Reply:
It is a variable of non-literal type. However long term plan is to make that work.